src/sinks/datadog/logs/service.rs

```diff
@@ -23,27 +22,27 @@ use vector_core::{
 
 use crate::{
     http::HttpClient,
+    sinks::datadog::ApiError,
     sinks::util::{retries::RetryLogic, Compression},
 };
 
 #[derive(Debug, Default, Clone)]
 pub struct LogApiRetry;
 
 impl RetryLogic for LogApiRetry {
-    type Error = LogApiError;
+    type Error = ApiError;
     type Response = LogApiResponse;
 
     fn is_retriable_error(&self, error: &Self::Error) -> bool {
         match *error {
-            LogApiError::HttpError { .. }
-            | LogApiError::BadRequest
-            | LogApiError::PayloadTooLarge => false,
+            ApiError::BadRequest | ApiError::PayloadTooLarge => false,
             // This retry logic will be expanded further, but specifically retrying unauthorized
-            // requests for now. I verified using `curl` that `403` is the respose code for this.
+            // requests and lower level HttpErrorsfor now.
+            // I verified using `curl` that `403` is the respose code for this.
             //
             // https://github.com/vectordotdev/vector/issues/10870
             // https://github.com/vectordotdev/vector/issues/12220
-            LogApiError::ServerError | LogApiError::Forbidden => true,
+            ApiError::HttpError { .. } | ApiError::ServerError | ApiError::Forbidden => true,
```

Comment:
I don't think we would want _all_ `ApiError::HttpError` cases to be retried. For example, `HttpError::BuildRequest` is a permanent error that will never work.

Reply:
I believe the type of error reported in the issue is `HttpError::CallRequest` (The error text seems to come from the hyper crate).
Last Friday I had concerns that all HttpError retries would be too broad :)  I had discussion  with @jszwedko  about it , he mentioned retrying any `HttpError` would be good but, perhaps he had not seen all the variants of `HttpError` ...
@fuchsnj  , would you recommend retrying any type of error here besides `HttpError::CallRequest` ?

Reply:
Addressed in latest commit.